src/ast/query.rs

```diff
@@ -174,6 +174,15 @@ impl SetExpr {
             None
         }
     }
+
+    /// If this `SetExpr` is a `SELECT`, returns a mutable [`Select`].
+    pub fn as_select_mut(&mut self) -> Option<&mut Select> {
+        if let Self::Select(select) = self {
+            Some(&mut **select)
+        } else {
+            None
+        }
+    }
```

Comment:
```suggestion
```
I think we would want to avoid a public mut API that is only used for internal tests, if needed it would likely make more sense as a helper function in the test module